service/storage_service.cc

```diff
@@ -427,7 +427,7 @@ future<> storage_service::topology_state_load(cdc::generation_service& cdc_gen_s
                     co_await _sys_ks.local().update_tokens(ip, {});
                     co_await _sys_ks.local().update_peer_info(ip, "host_id", id.uuid());
                 }
-                update_topology(ip, rs);
+                update_topology(host_id, ip, rs);
```

Comment:
I'm uneasy about this, we set the host_id back to itself, but if we set it to something wrong nothing will work.

Would it be better to make the host_id parameter optional and only engage it when we actually do want to change?

Reply:
This method is rebuilding topology from scratch based on system.topology. If we don't set it here, it will be missing.

Reply:
ok